qiskit/transpiler/passes/optimization/optimize_1q_decomposition.py

```diff
@@ -79,7 +79,7 @@ def run(self, dag):
             #     The U3 decomposer is only going to emit a sequence of length 1 anyhow.
             if "u3" in self._target_basis and len(run) == 1 and isinstance(run[0].op, U3Gate):
                 # Toss U3 gates equivalent to the identity; there we get off easy.
-                if np.array_equal(run[0].op.to_matrix(), np.eye(2)):
+                if np.allclose(run[0].op.to_matrix(), np.eye(2), 1e-15, 0):
```

Comment:
Can't this just be done from the parameter values themselves without casting to a matrix, e.g. see if the small angle approximations are close to the desired values or not?

Reply:
That would definetely work too but I thought this would be slightly more readable. I am good with either way so it is up to you

Reply:
I think its easier to keep as is, since the u3 parametrization has a continuum of synonyms `U3(0, x, -x)` for the identity thus the condition would be non-trivial.